src/Compilers/Core/Portable/CommandLine/CommandLineArguments.cs

```diff
@@ -464,7 +465,8 @@ public IEnumerable<AnalyzerReference> ResolveAnalyzerReferences(IAnalyzerAssembl
             EventHandler<AnalyzerLoadFailureEventArgs> errorHandler = (o, e) =>
             {
                 var analyzerReference = o as AnalyzerFileReference;
-                DiagnosticInfo diagnostic;
+                RoslynDebug.Assert(analyzerReference is object);
```

Comment:
Why not just make the `as` cast above a real cast?

Reply:
Didn't want to change code, in case my understanding of the code was wrong (changing to a cast would throw).

---
In reply to: [377819607](https://github.com/dotnet/roslyn/pull/41063#discussion_r377819607) [](ancestors = 377819607)